src/sense.jl

```diff
@@ -0,0 +1,75 @@
+#
+# Copyright (c) 2022 Tobias Thummerer, Lars Mikelsons
+# Licensed under the MIT license. See LICENSE file in the project root for details.
+#
+
+# this file containes placeholders for sensitivity (AD) related functions,
+# if FMISensitivity is not loaded, these functions dispatch to neutral statements,
+# if FMISensitivity is loaded, these functions are overwritten.
+
+# check if scalar/vector is ForwardDiff.Dual
+function isdual(e)
+    return false 
+end
+
+# check if scalar/vector is ForwardDiff.Dual
+function istracked(e)
+    return false 
+end
+
+# check types (Tag, Variable, Number) of ForwardDiff.Dual scalar/vector
+function fd_eltypes(e)
+    @assert "fd_eltypes needs FMISensitivity to work. Use `import FMISensitivity`."
+end
+
+# overwrites a ForwardDiff.Dual in-place 
+# inheritates partials
+function sense_set!(dst::AbstractArray{<:Real}, src::AbstractArray{<:Real})
+    dst[:] = src
+    return nothing
+end
+
+# makes Reals from ForwardDiff.Dual scalar/vector
+function undual(e::AbstractArray)
+    return collect(undual(c) for c in e)
+end
+function undual(e::Tuple)
+    return (collect(undual(c) for c in e)...,)
+end
```

Comment:
Performance-wise this is not very nice: if `e` is an array of fmi2reals then this will duplicate that array on every call to undual, which causes major slowdown. (see https://github.com/ThummeTo/FMIImport.jl/pull/97/commits/728319bf7e35b0307adc0311883db2f88c96dd35). Similar comments for `untrack`.

Reply:
also: where are the definitions for `undual(e::ForwardDiff.Dual)` etc? I guess this would dispatch to `undual(::Any)` right?

Reply:
Thank you. Any suggestions on how to convert AD primitives non-allocating?